src/main/webapp/scripts/itinerary-overview.js

```diff
@@ -102,32 +164,30 @@ function geocodePlaceId(hotelID, placeIDArray) {
  */
 
 //Hard coded for test (Gonna make minor changes for final product)
-async function calculateRoute(hotel, waypointsArray) {
-    console.log("Calculating Optimal Route");
-
-    start = "5902 N President George Bush Hwy, Garland, TX 75044, USA";
-
-    let rawWaypoints = ["525 Talbert Dr, Plano, TX 75093, USA",
-        "2134 Zurek Ln, Heath, TX 75126, USA",
-        "4234 Maple Ave #2403, Dallas, TX 75219, USA",]
+function calculateRoute(hotel, waypointsArray) {
+    start = hotel; //"5902 N President George Bush Hwy, Garland, TX 75044, USA";
 
     waypoints = [];
     end = "";
 
-    let length = rawWaypoints.length;
+    let length = waypointsArray.length;
 
-    while(rawWaypoints.length > 1){
-        for(let i = 0; i < rawWaypoints.length; i++){
+    while(waypointsArray.length > 1){
+        for(let i = 0; i < waypointsArray.length; i++){
             if(waypoints.length !== (length - 1)){
-                waypoints.push(rawWaypoints[0])
-                rawWaypoints.splice(0, 1)
+                waypoints.push(waypointsArray[0]);
+                waypointsArray.splice(0, 1);
```

Comment:
Why delete elements of the waypoints array instead of just iterating through it?

Reply:
Because as of right now I just store one extra point in the Array and use that as my endpoint whenever I change it to find the longest duration from point A (hotel) to point B (waypoint) then I can just iterate through the array. This should answers the optimization question too.